crates/genesis/src/lib.rs

```diff
@@ -422,6 +423,10 @@ pub struct ChainConfig {
     /// Clique parameters.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub clique: Option<CliqueConfig>,
+
+    /// Additional fields specific to each chain.
+    #[serde(flatten, default)]
+    pub extra_fields: BTreeMap<String, json_value>,
```

Comment:
```suggestion
    pub extra_fields: BTreeMap<String, serde_json::Value>,
```
and move serde-json from dev to regular deps in cargo.toml